crypto/rand/rand_lib.c

```diff
@@ -196,6 +198,13 @@ size_t rand_drbg_get_entropy(RAND_DRBG *drbg,
             goto err;
         }
 
+        drbg->reseed_next_counter = tsan_load(&drbg->reseed_prop_counter);
+        if (drbg->reseed_next_counter) {
+            drbg->reseed_next_counter++;
+            if(!drbg->reseed_next_counter)
+                drbg->reseed_next_counter = 1;
+        }
```

Comment:
I think I can move this block to the caller, only the case parent != NULL need to stay.